core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala

```diff
@@ -289,13 +290,14 @@ private[ui] class AllJobsPage(parent: JobsTab) extends WebUIPage("") {
       val summary: NodeSeq =
         <div>
           <ul class="unstyled">
-            {if (parent.sc.isDefined) {
-              // Total duration is not meaningful unless the UI is live
-              <li>
-                <strong>Total Uptime: </strong>
-                {UIUtils.formatDuration(System.currentTimeMillis() - startTime)}
-              </li>
+            <li>
+              <strong>Total Uptime: </strong>
+              {if (parent.sc.isDefined) {
```

Comment:
This seems OK, but can it be even simpler?

`{UIUtils.formatDuration((if (parent.sc.isDefined) System.currentTimeMillis() else endTime) - startTime)}`

Or maybe pull out a `val` to simplify.

In any event LGTM in principle